pennylane/devices/qubit/apply_operation.py

```diff
@@ -260,17 +260,23 @@ def apply_mid_measure(
     """
     if is_state_batched:
         raise ValueError("MidMeasureMP cannot be applied to batched states.")
+    if not np.allclose(np.linalg.norm(state), 1.0):
+        mid_measurements[op] = 0
+        return np.zeros_like(state)
     wire = op.wires
     probs = qml.devices.qubit.measure(qml.probs(wire), state)
     sample = np.random.binomial(1, probs[1])
     mid_measurements[op] = sample
+    if op.postselect is not None and sample != op.postselect:
+        return np.zeros_like(state)
     axis = wire.toarray()[0]
     slices = [slice(None)] * qml.math.ndim(state)
     slices[axis] = int(not sample)
     state[tuple(slices)] = 0.0
     state_norm = np.linalg.norm(state)
     if state_norm < 1.0e-15:
-        raise ValueError("Cannot normalize projected state.")
+        mid_measurements[op] = 0
```

Comment:
If we don't care about this dict when postselection doesn't match the sample, there's not really a need to update it here or above.

Reply:
If we do not exit early in the caller function, then we still need to assign a value so that posterior `Conditional`s can be apply without throwing.